src/dicomweb_client/web.py

```diff
@@ -685,16 +685,23 @@ def _decode_multipart_message(
                 if stream:
                     logger.debug(f'decode message content chunk #{i}')
                 data += chunk
-                while delimiter in data:
+
+                prev_part_index = 0
+                while (
+                        delimiter_index := data.find(delimiter, prev_part_index)
+                ) >= 0:
```

Comment:
I would rather not introduce the walrus operator here as it is not compatible with older python versions. We will need to drop support for 3.7 and below at some point, but this doesn't seem like a good reason to.

```suggestion
                while True:
                    delimiter_index = data.find(delimiter, prev_part_index)
                    if delimiter_index < 0:
                        break
```